Sources/StreamChat/Models/MessagePinning.swift

```diff
@@ -8,14 +8,12 @@ import Foundation
 public struct MessagePinning: Equatable {
     /// The expiration date of the pinning. Infinite expiration in case it is `nil`.
     public let expirationDate: Date?
-
-    /// Private initialiser. The static initialisers should be used for better syntax and ease of use.
-    init(expirationDate: Date?) {
-        self.expirationDate = expirationDate
-    }
+    
+    /// The pinning date. Equals to current date by default.
+    public var pinnedAt = Date()
```

Comment:
I don't think this is a good idea. The `pinnedAt` should be separate from the `MessagePinning` information. What if you want to set an expiration date prior to the actual action of pinning the message? The `pinnedAt` should be created like it was before when the action is performed. Setting the expiration date can be done separately from the actual action.

Reply:
Yeah, I was thinking about the following way:
```swift
var pinning = MessagePinning.noExpiration
pinning.pinnedAt = Date().addingTimeInterval(-10)
messageController.pinMessage(pinning) {}
```

But that's an edge case when someone wants to use the pinning date different from `now`.

Reply:
Why `Date().addingTimeInterval(-10)` ? 

```swift
let pinning = MessagePinning.noExpiration
messageController.pinMessage(pinning)
```

☝️ This should be the API. The `pinnedAt = Date()` should be done automatically inside the `messageController.pinMessage(pinning)` since it is the actual action.

Reply:
All right, let's move it to `pinMessage` as a second param 👍

Reply:
✅ Done